### PR TITLE
A2.3 fetch display profile biography

### LINKED ISSUE #9: A2.2 Receive & Store Biography
Story: ﻿As a User (Mentor, Mentee, or Educator), I want to be able to provide a biography description to describe myself, my goals, & my interests.﻿
﻿
﻿﻿Details:
When a user edits a biography and clicks submit on profile edit, this information should be properly passed to the backend.

﻿Technical Information:﻿
- React
- Send Post call to Django API (Do not handle Django receive)
- May require an HTTP handler such as Axios
- On Click Event handling
- Should Work with other Biography Task Assignees handling the UI & Backend Receive.
- Full-Stack

### LINKED ISSUE #10: A2.3 Fetch & Display Profile Biography
Story: ﻿As a User (Mentor, Mentee, or Educator), I want to be able to provide a biography description to describe myself, my goals, & my interests.﻿
﻿
﻿﻿Details:
Biography belonging to the user should be desplayed.

﻿Technical Information:﻿
- React
- Fetch Biography with GET HTTP API Call
- Display Biography
- Full-Stack

### PR DESCRIPTION
**Pull request by tdang2180**
fixes #9 and resolves #10 
- Implemented a fetch method using Axios instance to get data from the user's biography and display it.
- Fixed the save biography method 
- Fixed post-call handling for biography 

I thought I picked an easy task to learn Axios but this was harder than expected lol.

### CODE REVIEW

File: mesh/profiles/views.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def get(self, request, account_id, *args, **kwargs):
         try:
             profile = Profile.objects.get(accountID=account_id)
             profile_biography = profile.biography
-            return JsonResponse({"message": profile_biography}, safe=False, status=200)
+            return JsonResponse({"biography": profile_biography}, safe=False, status=200)
```
Comment (JRSiwiecki) [inline mesh/profiles/views.py:43]: Is the `safe = False` flag necessary here?

File: meshapp/package-lock.json
Comment (JRSiwiecki) [inline meshapp/package-lock.json:1]: Why was the package-lock.json updated?

File: mesh/profiles/views.py [hunk 2]
```diff
@@ -348,22 +338,24 @@ def post_data(account_id, name, request):
     """
     try: 
         profile = Profile.objects.get(accountID = account_id)
-        data = request.POST[name]   
+        data = json.loads(request.body)[name]
```
Comment (JRSiwiecki) [inline mesh/profiles/views.py:341]: Good use of existing functions 👍🏼

File: meshapp/src/profile/profile-textfield.tsx [hunk 3]
```diff
@@ -22,7 +22,7 @@ const ProfileTextField = (props: {
   placeholder: string;
   text: string;
   charLimit: number;
-  handleSave: any;
+  handleSave: (text: string)=> void;
```
Comment (JRSiwiecki) [inline meshapp/src/profile/profile-textfield.tsx:25]: Just out of curiosity, why was this changed?

**Review by JRSiwiecki — COMMENTED** [review]
This looks pretty good for the most part, just some small things and questions I have. Re-ping me on discord after you've had a chance to look at my review.

File: meshapp/src/profile/profile-textfield.tsx (hunk 3 above)
Comment (tdang2180) [reply meshapp/src/profile/profile-textfield.tsx:25]: The method still works with type any, but TypeScript says it's best practice to clarify a specific type unless the data type is unknown.

File: meshapp/package-lock.json
Comment (tdang2180) [reply meshapp/package-lock.json:1]: I reverted it. I think it was updated because I did npm install.

**Review by JRSiwiecki — APPROVED** [review]
👍🏼 good work!

File: mesh/profiles/views.py [hunk 4]
```diff
@@ -56,17 +56,7 @@ def post(self, request, account_id, *args, **kwargs):
...
-      if 'accountID' not in data:
-          return JsonResponse({'error': 'Invalid request. Missing accountID field.'}, status=401)
-      if 'biography' not in data:
-          return JsonResponse({'error': 'Invalid request. Missing biography field.'}, status=400)
-      else:
-          user = Profile.objects.get(accountID=account_id)
-          user.biography = data["biography"]
-          user.save()
-          return JsonResponse({'message': "Biography saved successfully."}, status=200)
+      return post_data(account_id, "biography", request)
```
Comment (BradJKim) [inline mesh/profiles/views.py:59]: I'm aware that the post request for biography was outdated, does that means we need to make a new task for handling to post request for biography?

Comment (tdang2180) [reply mesh/profiles/views.py:59]: No, I fixed the post request for biography in this PR.

**Review by BradJKim — APPROVED** [review]
